pagopa/io-app
# feat: [IOCOM-857] Payment data on message details

## Short description
This PR adds the payment data and its actualisation to the new DS message details screen
![Simulator Screenshot - iPhone 15 - 2024-03-06 at 18 02 46](https://github.com/pagopa/io-app/assets/5150343/50bf1671-a769-4935-a025-ec53d42ee28c)

## List of changes proposed in this pull request
- `MessagePaymentItem` moved from SEND to MESSAGES. Its business logic it is now generic
- New `MessageDetailsPayment` component to display a standard message payment (using existing actualisation logic in `MessagePaymentItem`)
- `MessageDetailsScreen` updated to include the new component and to check on focus it there is a previously selected payment to update (related to its payments - this is needed to ensure that, when a user pays, the status is updated inside the message view upon re-entering)
- Actualisation logic has been changed from handling a single payment to remembering and handling a set of payments, since after having paid, the navigation brings the user to the wallet home (instead of the message details as in SEND). In this case, the user may open another message and pay its payment. Upon returning to the first message, the payment must be updated (and also the second one, if the user also goes to that message screen)
- SEND components and screens have been ported to the new business logic, both for shared components and payment actualisation
- `isProfileEmailValidatedSelector` and `isPagoPaSupportedSelector` selectors have been re-written in order to avoid using an unnecessary `createSelector` caching
- Proper typing of returned data for `configureStoreAndPersistor`

## How to test
Using the io-dev-api-server, check that payment data in a message (with payment) is properly shown and actualised. Check that it is payable and that, upon returning to the message details, it gets updated. Check also for its different failure statuses.
For retro-compatibility, check that SEND payments still work.
Also, try paying a lot of differente messages' payments without going back into the message upon payment. After having paid them all, go back to each message details and check that every payment is being updated.
